src/siotls/configuration.py

```diff
@@ -0,0 +1,57 @@
+import dataclasses
+import typing
+from siotls.iana import (
+    CipherSuites,
+    SignatureScheme,
+    NamedGroup,
+    ALPNProtocol,
+)
+
+
+@dataclasses.dataclass()
+class TLSConfiguration:
+    side: typing.Literal['client', 'server']
+
+    # mandatory
+    cipher_suites: list[CipherSuites | int] = \
+        dataclasses.field(default_factory=[
+            CipherSuites.TLS_CHACHA20_POLY1305_SHA256,
+            CipherSuites.TLS_AES_256_GCM_SHA384,
+            CipherSuites.TLS_AES_128_GCM_SHA256,
+        ].copy)
+    digital_signatures: list[SignatureScheme | int] = \
+        dataclasses.field(default_factory=[
+            SignatureScheme.rsa_pkcs1_sha256,
+            SignatureScheme.rsa_pss_rsae_sha256,
+            SignatureScheme.ecdsa_secp256r1_sha256,
+        ].copy)
+    key_exchanges: list[NamedGroup | int] = \
+        dataclasses.field(default_factory=[
+            NamedGroup.x25519,
+            NamedGroup.secp256r1,
+        ].copy)
```

Comment:
To be consistent with the NegociatedConfig that is typed without `int` and because we sure would need to update siotls to support extra unknown ciphers

```suggestion
    # mandatory
    cipher_suites: list[CipherSuites] = \
        dataclasses.field(default_factory=[
            CipherSuites.TLS_CHACHA20_POLY1305_SHA256,
            CipherSuites.TLS_AES_256_GCM_SHA384,
            CipherSuites.TLS_AES_128_GCM_SHA256,
        ].copy)
    digital_signatures: list[SignatureScheme] = \
        dataclasses.field(default_factory=[
            SignatureScheme.rsa_pkcs1_sha256,
            SignatureScheme.rsa_pss_rsae_sha256,
            SignatureScheme.ecdsa_secp256r1_sha256,
        ].copy)
    key_exchanges: list[NamedGroup] = \
        dataclasses.field(default_factory=[
            NamedGroup.x25519,
            NamedGroup.secp256r1,
        ].copy)
```